src/deadline/cinema4d_adaptor/Cinema4DAdaptor/adaptor.py

```diff
@@ -192,7 +192,7 @@ def _get_regex_callbacks(self) -> list[RegexCallback]:
             completed_regexes = [re.compile(".*Finished Rendering.*")]
             progress_regexes = [re.compile(".*Progress ([0-9]+)%.*")]
             error_regexes = [
-                re.compile(".*Error: .*|.*\\[Error\\].*|.*CRITICAL: .*", re.IGNORECASE)
+                re.compile(".*Error: .*|.*\\[Error\\].*|.*CRITICAL: Stop.*", re.IGNORECASE)
```

Comment:
Can we use a negative lookahead for `CRITICAL: nullptr` messages instead of just looking for `CRITICAL: Stop` messages?

Also, as a general note, should we use a raw string (often used with regexes) and put parenthesis around each of the three cases to make the regexes easier to understand?

Reply:
Yeah, I touched upon this in the description : https://amzn-aws.slack.com/archives/C02SZF1776F/p1730928489056379 and I wasn't sure if we should check for negative look ahead. 

I couldn't find any information regarding which critical errors should fail the job for Cinema 4D. Hence, I'm only checking for stops.
If we find out that its not true and somehow CRITICAL: nullptr is special, we can change the error regex accordingly in the future.

Reply:
> Also, as a general note, should we use a raw string (often used with regexes) and put parenthesis around each of the three cases to make the regexes easier to understand?

How about I add this as a follow up? This is one of the fixes that we need to get in as quickly as possible to ensure that we don't have customers that have successful jobs that fail because of this.

Reply:
Discussed with Karthik. Cinema4D uses a different error convention than other software. Approving as this is the most reasonable regex we can come up with right now and can always update it if we find additional cases for which to fail the render.

Reply:
> should we use a raw string (often used with regexes) and put parenthesis around each of the three cases to make the regexes easier to understand?

Looks like this PR was not reviewed by others yet so I might as well update it. So I updated according to your suggestions. :)